### DISCUSSION TITLE
Same model, same number of rows - but slightly different data set leads to out of memory

### DISCUSSION THREAD

**Discussion by jkginfinite** [moj-analytical-services/splink#1607] [Q&A]
I have trained a splink model on roughly 8 million rows, and it predicts results well. It contains a first name column as a feature. I saved the model.

I then used my saved model on a different 8 million rows of data - only this time the data ONLY consists of first names with the letter "J".

For some odd reason, it runs out of memory when the first names are concentrated to only names of a single letter (can be another letter... for example "A"... Alfred, Adam, Andy..)

Doing this task on an AWS EMR pyspark system.

How can this happen and what does it mean?

**Comment by RobinL** [comment]
I think this is probably to be expected. If you're using first name as a blocking variable but you make every first name the letter j then effectively you're no longer blocking on first name and the number of comparisons will rise a huge amount
